src/biogeophys/SoilHydrologyMod.F90

```diff
@@ -568,7 +614,32 @@ subroutine QflxH2osfcDrain(bounds, num_hydrologyc, filter_hydrologyc, &
            qflx_h2osfc_drain(c) = h2osfc(c)/dtime
            truncate_h2osfc_to_zero(c) = .true.
         else
-           qflx_h2osfc_drain(c)=min(frac_h2osfc(c)*qinmax(c),h2osfc(c)/dtime)
+
+           ! define maximum drainage
+           ! NOTE: check fraction
+           drainMax = frac_h2osfc(c)*qinmax(c)
+
+           ! switch between different numerical solutions
+           select case(ixSolution)
+
+              ! constrained Explicit Euler solution with operator splitting (original)
+              case(ixExplicitEuler)
+                 qflx_h2osfc_drain(c)=min(drainMax,h2osfc(c)/dtime)
+
+              ! implicit Euler solution with operator splitting
+              case(ixImplicitEuler)
+                 funcName=>drainPond
+                 call implicitEuler(funcName, dtime, h2osfc(c), h2osfc1, err, message)
+                 if(err/=0) call endrun(subname // ':: '//trim(message))
+                 call drainPond(h2osfc1, qflx_h2osfc_drain(c))
```

Comment:
**Comment from @billsacks Sep 7, 2017**

Is this a general pattern - that, to get the final flux with implicit Euler, we do a final function call like this? If so, can this final call be incorporated into the implicitEuler routine itself?

Reply:
**Comment from @martynpclark Sep 7, 2017**

Yes. The final flux should be a function of the state at the end of the time step.

Reply:
**Comment from @billsacks Sep 11, 2017**

Addressed in NCAR/clm-ctsm#14 in NCAR/clm-ctsm@315c14b